### PR TITLE
Bug 1331538 - Error when trying to save RemoteConnectionFactory confi…

### PR DESCRIPTION
**Pull request by josejulio**
…guration

 Removed property protocol-manager-factory as currently is not working https://issues.jboss.org/browse/JBEAP-4506
 Checks if the (Profile) sufix exists.
 Skips some checks on JMS Queue and JMS Topic
 Changed to readonly two attributes of JMS Queue.

### CODE REVIEW

File: modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/BaseComponent.java [hunk 1]
```diff
@@ -952,4 +954,14 @@ public static ComplexRequest create(String requestName) {
     public static Log getLog() {
         return LOG;
     }
+
+    public static String resourceTypeNameByRemovingProfileSuffix(String resourceTypeName) {
+        if (resourceTypeName.length() > PROFILE_SUFFIX.length() && resourceTypeName
```
Comment (burmanm) [inline modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/BaseComponent.java]: Why not just ``resourceTypeName.indexOf(PROFILE_SUFFIX) != -1`` ?

File: modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/ConnectorDiscoveryGroupValidatorComponent.java [hunk 2]
```diff
@@ -151,9 +149,13 @@ String getConnectorPropertyName() {
...
         @Override
         boolean isConnectorPropertyDefined() {
-            PropertyMap connector = configuration.getMap(CONNECTOR_PROPERTY);
-            if (connector != null) {
-                return isNotBlank(connector.getSimpleValue("name:0", EMPTY_STRING));
+            PropertyList connector = configuration.getList(CONNECTOR_PROPERTY);
+            if (connector != null && connector.getList().size() > 0) {
+                Property property = connector.getList().get(0);
+                if (property instanceof PropertySimple) {
+                    PropertySimple propertySimple = (PropertySimple) connector.getList().get(0);
```
Comment (burmanm) [inline modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/ConnectorDiscoveryGroupValidatorComponent.java]: ``(PropertySimple) property;`` ?

File: modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/ConnectorDiscoveryGroupValidatorComponent.java [hunk 3]
```diff
@@ -172,8 +174,17 @@ public Operation getBatchOperation() {
...
-                compositeOperation.addStep(new WriteAttribute(address, CONNECTOR_ATTRIBUTE, Collections.singletonMap(
-                    configuration.getMap(CONNECTOR_PROPERTY).getSimpleValue("name:0", EMPTY_STRING), null)));
+                List<Property> propertyList = configuration.getList(CONNECTOR_PROPERTY).getList();
+                List<String> connectors = new ArrayList<String>(propertyList.size());
+                for (Property property : propertyList) {
+                    if (property instanceof PropertySimple) {
+                        PropertySimple propertySimple = (PropertySimple) property;
+                        connectors.add(propertySimple.getStringValue());
+                    } else {
+                        getLog().warn(property.getName() + " property has unexpected type: " + property.getClass());
```
Comment (burmanm) [inline modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/ConnectorDiscoveryGroupValidatorComponent.java]: Is this warning logging of any use? Should you throw Exception or just ignore the thing? Or log in debug mode at most?

File: modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/ConnectorDiscoveryGroupValidatorComponent.java (hunk 2 above)
Comment (josejulio) [inline modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/ConnectorDiscoveryGroupValidatorComponent.java]: Yep, it should read that :+1:.

File: modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/BaseComponent.java (hunk 1 above)
Comment (josejulio) [inline modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/BaseComponent.java]: On a second though, `resourceTypeName.endsWith(PROFILE_SUFFIX)` would work better.

**Pull request merged**